### DISCUSSION TITLE
How to manage simultaneous use of multiple displays with different drivers?

### DISCUSSION THREAD

**Discussion by michcfr** [Bodmer/TFT_eSPI#1458] [Compatible displays and setup files]
Hello
I want to connect my ESP32 to three displays, used simultaneously, each with a different driver: ST7789, ILI9486, GC9A01.
How can I manage three configurations (compile with three User_Setup_Select.h, one for each driver) for the three different displays?

Regards,
Michel

**Comment by michcfr** [comment]
> Hello I want to connect my ESP32 to three displays, used simultaneously, each with a different driver: ST7789, ILI9486, GC9A01. How can I manage three configurations (compile with three User_Setup_Select.h, one for each driver) for the three different displays?
> 
> Regards, Michel

Up to myself.
If somebody Can help. Thx

**Reply by FarazFKhan** [reply]
@michcfr Hi there, I am looking for the same information. Did you end up figuring out how to support different driver based displays?

**Reply by michcfr** [reply]
Hello, 
I switched to Arduino_GFX: https://github.com/moononournation/Arduino_GFX
Regards

**Reply by FarazFKhan** [reply]
Ahhh, I see. Thank you for letting me know about that. I will investigate that today at some point. 👍

**Comment by FarazFKhan** [comment]
One way in my mind was to run the same library with a different library name & variable names. So I turned all 8357 mentions of "TFT" to "TFT2" and all 5861 mentions of "tft" to "tft2". This however did not work, unfortunately. 

If anyone is able to assist with how to run the same library twice on a single esp32 so that I can use each instance of the library to run a specific TFT driver, that help would very much be appreciated!

I look forward to a solution 😀

**Reply by Frank-Bemelman** [reply]
Hmm, that would require a nice amount of fiddling. I think I would start with making 3 identical copies of the library, in different folders.
Rename the .h and .cpp files by adding a an extra letter, like A, B, C.
Then rename the class references (in .h and .cpp) from TFT_eSPI to TFT_eSPIA, TFT_eSPIB and TFT_ESPIC. 
Select the driver in each belonging User_SetupA.h and so on, and give them the right definitions of which pins uses. You could share the SPI bus, if you manually set the proper CS output for each display, before you call any functions that does something with that display.
Not sure if you will find other pitfalls. 
Wish you luck ;-)
